Automattic/newspack-plugin
# feat(salesforce): manually sync orders from WC order UI

### All Submissions:






### Changes proposed in this Pull Request:



~Not to merged before #1429.~ Ready for review! Extends the WooCommerce admin UI for orders to more conveniently show the orders' sync status with Salesforce, as well as allow admins to manually sync or resync single orders with a button click.

### How to test the changes in this Pull Request:

1. Set up a Salesforce connection on your test site (follow instructions on [this page](https://newspack.pub/support/reader-revenue/) to set up a connection, and DM me if you need a Salesforce dev account).
2. Make a test donation as a reader and wait a few minutes.
3. In WP admin > WooCommerce > Orders, find the order corresponding to your test donation.
4. In the Order Actions metabox, verify that you see a new "Salesforce sync status" message with the sync status of your order (will only appear if you have an active Salesforce connection). Also verify that if the order shows a "synced" status, that it's clickable to the Salesforce Opportunity record:

<img width="285" alt="Screen Shot 2022-01-07 at 11 34 19 AM" src="https://user-images.githubusercontent.com/2230142/148590549-3ea75689-f1b2-4774-86aa-45a899dabc26.png">

5. Also observe that on successful or failed sync, a message is added to the "Order notes" metabox (this is to add some slightly more user-friendly sync logging outside of the WooCommerce webhook logs):

<img width="287" alt="Screen Shot 2022-01-07 at 11 34 56 AM" src="https://user-images.githubusercontent.com/2230142/148590636-4beaa75a-e85b-45b0-a64a-68d357ac0b10.png">

7. In the Order Actions metabox, expand the "Choose an action..." dropdown. Confirm that there's a new action to "Sync order to Salesforce" (will only appear if you have an active Salesforce connection):

<img width="309" alt="Screen Shot 2022-01-07 at 11 36 04 AM" src="https://user-images.githubusercontent.com/2230142/148590767-40bcd06b-ee45-4e19-b07d-2cb8762edf5f.png">

8. Select the new option and click the [>] button to execute the action. For already-synced orders, this will resync the order to the existing Opportunity record in Salesforce. Only billing info and close date will be updated, as the other order data is not updateable in WooCommerce. For orders that have not been synced for whatever reason, it will create a new Opportunity record in Salesforce.
9. Log into Salesforce and confirm that the manual sync action successfully updated or created the opportunity record.

### Other information:
